game/engine.py

```
@@ -116,11 +116,13 @@ def generate_pgn(self, white_name='White', black_name='Black'):
             result = '1-0' if self.current_turn == 'black' else '0-1'
 
         pgn_moves = []
+        def _fix_castling(move):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `_fix_castling`

(ANN202)